fedimint-core/src/db/mod.rs

```diff
@@ -811,32 +827,43 @@ impl<'parent> DatabaseTransaction<'parent> {
         return self.tx.commit_tx().await;
     }
 
+    pub async fn expect_commit_tx(mut self) {
```

Comment:
If I was brand new to this project, it would be a little weird to call `expect_commit_tx` to commit to the database. I would probably reach for `commit_tx` because it sounds more like what I'm looking to do.

I wonder if this function could just be `commit_tx` and the current `commit_tx` could be `commit_tx_result` or something? 

Also thinking that the one that returns a result could probably be private now? But seems like we use it for test assertions so maybe not ...